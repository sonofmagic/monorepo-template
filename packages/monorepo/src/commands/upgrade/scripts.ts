/**
 * 升级时注入到 package.json 的脚本命令集合，保证常用脚本齐全。
 */
export const scripts = {
  'script:init': 'monorepo init',
  'script:sync': 'monorepo sync',
  'script:clean': 'monorepo clean',
  'script:mirror': 'monorepo mirror',
  'commitlint': 'commitlint --edit',
}

export const scriptsEntries = Object.entries(scripts)
