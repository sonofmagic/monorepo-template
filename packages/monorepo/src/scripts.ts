export const scripts = {
  'script:init': 'monorepo init',
  'script:sync': 'monorepo sync',
  'script:clean': 'monorepo clean',
  'script:mirror': 'monorepo mirror',
  'commitlint': 'commitlint --edit',
}

export const scriptsEntries = Object.entries(scripts)
