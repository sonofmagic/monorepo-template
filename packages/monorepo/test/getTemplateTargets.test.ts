import { getTemplateTargets } from '../scripts/getTemplateTargets'

describe('getTemplateTargets', () => {
  it('should work', async () => {
    const targets = await getTemplateTargets()
    expect(targets).toMatchSnapshot()
  })
})
