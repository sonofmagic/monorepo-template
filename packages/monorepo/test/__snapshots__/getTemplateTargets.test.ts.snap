// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getTemplateTargets > should work 1`] = `
[
  "packages/tsup-template/build.config.ts",
  "packages/tsup-template/package.json",
  "packages/tsup-template/src/index.ts",
  "packages/tsup-template/test/index.test.ts",
  "packages/tsup-template/tsconfig.json",
  "packages/tsup-template/tsup.config.ts",
  "packages/tsup-template/vitest.config.ts",
  "packages/unbuild-template/build.config.ts",
  "packages/unbuild-template/package.json",
  "packages/unbuild-template/src/index.ts",
  "packages/unbuild-template/src/utils.ts",
  "packages/unbuild-template/test/index.test.ts",
  "packages/unbuild-template/tsconfig.json",
  "packages/unbuild-template/vitest.config.ts",
  "packages/vue-lib-template/.gitignore",
  "packages/vue-lib-template/eslint.config.js",
  "packages/vue-lib-template/index.html",
  "packages/vue-lib-template/lib/HelloWorld.vue",
  "packages/vue-lib-template/lib/index.ts",
  "packages/vue-lib-template/package.json",
  "packages/vue-lib-template/src/App.vue",
  "packages/vue-lib-template/src/main.ts",
  "packages/vue-lib-template/src/pages/index.vue",
  "packages/vue-lib-template/src/router/index.ts",
  "packages/vue-lib-template/src/style.css",
  "packages/vue-lib-template/src/vite-env.d.ts",
  "packages/vue-lib-template/test/index.test.ts",
  "packages/vue-lib-template/tsconfig.app.json",
  "packages/vue-lib-template/tsconfig.json",
  "packages/vue-lib-template/tsconfig.node.json",
  "packages/vue-lib-template/tsconfig.test.json",
  "packages/vue-lib-template/vite.config.ts",
  "packages/vue-lib-template/vite.shared.config.ts",
  "packages/vue-lib-template/vitest.config.ts",
  "apps/server/.gitignore",
  "apps/server/package.json",
  "apps/server/src/app.ts",
  "apps/server/src/config.ts",
  "apps/server/src/fetch-entry.ts",
  "apps/server/src/node-entry.ts",
  "apps/server/test/index.test.ts",
  "apps/server/tsconfig.json",
  "apps/server/tsup.config.ts",
  "apps/server/vitest.config.ts",
  "apps/server/wrangler.jsonc",
  "apps/client/.gitignore",
  "apps/client/eslint.config.js",
  "apps/client/index.html",
  "apps/client/package.json",
  "apps/client/public/vite.svg",
  "apps/client/src/App.vue",
  "apps/client/src/main.ts",
  "apps/client/src/pages/index.vue",
  "apps/client/src/router/index.ts",
  "apps/client/src/stores/i18n.ts",
  "apps/client/src/stores/index.ts",
  "apps/client/src/style.css",
  "apps/client/src/trpc/index.ts",
  "apps/client/src/vite-env.d.ts",
  "apps/client/tsconfig.app.json",
  "apps/client/tsconfig.json",
  "apps/client/tsconfig.node.json",
  "apps/client/tsconfig.worker.json",
  "apps/client/vite.config.ts",
  "apps/client/worker/app.ts",
  "apps/client/worker/config.ts",
  "apps/client/worker/fetch-entry.ts",
  "apps/client/worker/node-entry.ts",
  "apps/client/worker/trpc-router.ts",
  "apps/client/wrangler.jsonc",
  "apps/website/.gitignore",
  "apps/website/.vitepress/config.ts",
  "apps/website/.vitepress/theme/index.ts",
  "apps/website/.vitepress/theme/tailwind.css",
  "apps/website/index.md",
  "apps/website/monorepo/index.md",
  "apps/website/monorepo/manage.md",
  "apps/website/monorepo/publish.md",
  "apps/website/package.json",
  "apps/website/public/logo.jpg",
  "apps/website/thinking.md",
  "apps/website/tools/changeset.md",
  "apps/website/tools/husky.md",
  "apps/website/tools/lint-staged.md",
  "apps/website/tools/pnpm.md",
  "apps/website/tools/renovate.md",
  "apps/website/tools/turborepo.md",
  "apps/website/why/assets/npm-dt.svg",
  "apps/website/why/assets/npm-ts.svg",
  "apps/website/why/examples/0.npm-basic-package/index.js",
  "apps/website/why/examples/0.npm-basic-package/package.json",
  "apps/website/why/examples/1.npm-basic-package/index.cjs",
  "apps/website/why/examples/1.npm-basic-package/index.d.ts",
  "apps/website/why/examples/1.npm-basic-package/index.js",
  "apps/website/why/examples/1.npm-basic-package/package.json",
  "apps/website/why/examples/2.npm-basic-package/README.md",
  "apps/website/why/examples/2.npm-basic-package/package-lock.json",
  "apps/website/why/examples/2.npm-basic-package/package.json",
  "apps/website/why/examples/2.npm-basic-package/rename-ext.js",
  "apps/website/why/examples/2.npm-basic-package/src/index.ts",
  "apps/website/why/examples/2.npm-basic-package/tsconfig.cjs.json",
  "apps/website/why/examples/2.npm-basic-package/tsconfig.esm.json",
  "apps/website/why/examples/2.npm-basic-package/tsconfig.json",
  "apps/website/why/examples/3.npm-basic-package/README.md",
  "apps/website/why/examples/3.npm-basic-package/build.config.ts",
  "apps/website/why/examples/3.npm-basic-package/package-lock.json",
  "apps/website/why/examples/3.npm-basic-package/package.json",
  "apps/website/why/examples/3.npm-basic-package/src/index.ts",
  "apps/website/why/examples/3.npm-basic-package/tsconfig.json",
  "apps/website/why/examples/3.npm-basic-package/tsup.config.ts",
  "apps/website/why/examples/usage-cjs/cjs.cjs",
  "apps/website/why/examples/usage-cjs/esm.mjs",
  "apps/website/why/examples/usage-cjs/index.js",
  "apps/website/why/examples/usage-cjs/package-lock.json",
  "apps/website/why/examples/usage-cjs/package.json",
  "apps/website/why/examples/usage-esm/cjs.cjs",
  "apps/website/why/examples/usage-esm/createRequire.js",
  "apps/website/why/examples/usage-esm/esm.mjs",
  "apps/website/why/examples/usage-esm/index.js",
  "apps/website/why/examples/usage-esm/package-lock.json",
  "apps/website/why/examples/usage-esm/package.json",
  "apps/website/why/how-to-reuse-js-code.md",
  "apps/website/why/index.md",
  "apps/website/why/js-cjs-mjs.md",
  "apps/website/why/js-keywords.md",
  "apps/website/why/modern/bundlers.md",
  "apps/website/why/modern/dts.md",
  "apps/website/why/modern/esm-vs-cjs.md",
  "apps/website/why/modern/index.md",
  "apps/website/why/modern/package-entry-points.md",
  "apps/website/why/modern/typescript.md",
  "apps/website/why/publish-basic-npm-package.md",
  "apps/website/why/what-is-npm-package.md",
  "apps/cli/.gitignore",
  "apps/cli/bin/index.js",
  "apps/cli/dev/index.ts",
  "apps/cli/package.json",
  "apps/cli/src/cli.ts",
  "apps/cli/src/index.ts",
  "apps/cli/tsconfig.json",
  "apps/cli/tsup.config.ts",
]
`;
